pkgs/development/libraries/gvfs/default.nix

```diff
@@ -108,6 +108,8 @@ stdenv.mkDerivation rec {
     "-Dkeyring=false"
     "-Dhttp=false"
     "-Dgoogle=false"
+  ] ++ lib.optionals (avahi == null) [
+    "-Ddnssd=false"
   ] ++ lib.optionals (samba == null) [
```

Comment:
I know that this pattern is discouraged, but I don't really want to crawl through all the packages and search where `samba=null` is used and replace both...